Dockerfile

```diff
@@ -1,4 +1,38 @@
-FROM node:21.7.3-bookworm-slim
+FROM node:21.7.3-bookworm-slim as base
+
+# Prerequisites
+
+# renovate: datasource=repology depName=debian_12/ca-certificates versioning=loose
+ENV CACERTIFICATES_VERSION=20230311
+
+# Ca-Certificates is required for connection to Azure DevOps
+RUN apt-get update -y && \
+    apt-get install -y --no-install-recommends ca-certificates=${CACERTIFICATES_VERSION} && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
+FROM base as build
+
+# Make sure to fail due to an error at any stage in shell pipes
+SHELL ["/bin/bash", "-o", "pipefail", "-c"]
+
+# renovate: datasource=repology depName=debian_12/curl versioning=deb
+ENV CURL_VERSION=7.88.1-10+deb12u5
+
+RUN apt-get update -y && \
+    # Install necessary dependencies
+    apt-get install -y --no-install-recommends curl=${CURL_VERSION}
+
+# Install the Flux CLI
+
+# renovate: datasource=github-releases depName=fluxcd/flux2 extractVersion=^v(?<version>.*)$
+ENV FLUX_VERSION=2.2.3
+
+RUN curl -s https://fluxcd.io/install.sh | FLUX_VERSION=${FLUX_VERSION} bash && \
+    # Smoke test    
+    flux --version
```

Comment:
@rsitrifork Smoke test should run in the final image (after binaries have been copied over), not the builder image, to test if the tool actually works there or if for example some dependencies are missing.